src/sanitizer.js

```diff
@@ -241,3 +249,83 @@ export function isValidAttr(attrName, attrValue) {
 
   return true;
 }
+
+/**
+ * Resolves the attribute value. The main purpose is to rewrite URLs as
+ * described in `resolveUrlAttr`.
+ * @param {string} tagName
+ * @param {string} attrName
+ * @param {string} attrValue
+ * @return {string}
+ */
+function resolveAttrValue(tagName, attrName, attrValue) {
+  if (attrName == 'src' || attrName == 'href' || attrName == 'srcset') {
+    return resolveUrlAttr(tagName, attrName, attrValue, window.location);
+  }
+  return attrValue;
+}
+
+/**
+ * Rewrites the URL attribute values. URLs are rewritten as following:
+ * - If URL is absolute, it is not rewritten
+ * - If URL is relative, it's rewritten as absolute against the source origin
+ * - If resulting URL is a `http:` URL and it's for image, the URL is rewritten
+ *   again to be served with AMP Cache (cdn.ampproject.org).
+ *
+ * @param {string} tagName
+ * @param {string} attrName
+ * @param {string} attrValue
+ * @param {!Location} windowLocation
+ * @return {string}
+ * @private Visible for testing.
+ */
+export function resolveUrlAttr(tagName, attrName, attrValue, windowLocation) {
+  const isProxy = isProxyOrigin(windowLocation);
+  const baseUrl = parseUrl(getSourceUrl(windowLocation));
+
+  if (attrName == 'href' && attrValue.indexOf('#') != 0) {
+    return resolveRelativeUrl(attrValue, baseUrl);
+  }
+
+  if (attrName == 'src') {
+    if (tagName == 'amp-img') {
+      return resolveImageUrlAttr(attrValue, baseUrl, isProxy);
+    }
+    return resolveRelativeUrl(attrValue, baseUrl);
+  }
+
+  if (attrName == 'srcset') {
+    let srcset;
+    try {
+      srcset = parseSrcset(attrValue);
+    } catch (e) {
+      setTimeout(() => {throw e;});
+      return attrValue;
```

Comment:
It seems wrong to blacklist the whole response because `srcset` might be broken. Our elements catch and report these problems in a non-blocking way.